From 3cd151148381257ad90bf48ebeb464694ce06c0f Mon Sep 17 00:00:00 2001
From: Anton Bambura <jenneron@protonmail.com>
Date: Tue, 19 Jan 2021 17:43:37 +0200
Subject: [PATCH 171/210] arm64: dts: qcom: msm8916-wingtech-wt865x8: add bms
 and charging

This patch adds support for charging and measuring battery status. These
devices have same battery and use bms for measuring battery status.
Lenovo A6000 uses PM8916 charger. Lenovo A6010 uses external charger
which does not have a driver in mainline yet.

Signed-off-by: Anton Bambura <jenneron@protonmail.com>
---
 .../dts/qcom/msm8916-wingtech-wt86518.dts     | 21 ++++++++++++-----
 .../dts/qcom/msm8916-wingtech-wt865x8.dtsi    | 23 +++++++++++++++++++
 2 files changed, 38 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts b/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts
index b010c24c2ae..9237cc168d8 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts
@@ -10,10 +10,6 @@ / {
 	chassis-type = "handset";
 };
 
-&pm8916_usbin {
-	status = "okay";
-};
-
 &blsp_i2c2 {
 	accelerometer@e {
 		compatible = "kionix,kxcj91008";
@@ -28,6 +24,19 @@ accelerometer@e {
 	};
 };
 
+&pm8916_charger {
+	status = "okay";
+
+	qcom,vdd-safe = <4300000>;
+	qcom,ibat-safe = <900000>;
+
+	monitored-battery = <&battery>;
+};
+
+&pm8916_bms {
+	power-supplies = <&pm8916_charger>;
+};
+
 &reg_lcd_neg {
 	regulator-min-microvolt = <5400000>;
 	regulator-max-microvolt = <5400000>;
@@ -40,9 +49,9 @@ &reg_lcd_pos {
 
 &usb {
 	dr_mode = "peripheral";
-	extcon = <&pm8916_usbin>;
+	extcon = <&pm8916_charger>;
 };
 
 &usb_hs_phy {
-	extcon = <&pm8916_usbin>;
+	extcon = <&pm8916_charger>;
 };
diff --git a/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8.dtsi b/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8.dtsi
index 7c1fd0e01ff..d2659627d56 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8.dtsi
@@ -22,6 +22,24 @@ backlight: backlight {
 		default-brightness-level = <255>;
 	};
 
+	battery: battery {
+		compatible = "simple-battery";
+		voltage-min-design-microvolt = <3400000>;
+		voltage-max-design-microvolt = <4350000>;
+		energy-full-design-microwatt-hours = <8740000>;
+		charge-full-design-microamp-hours = <2300000>;
+
+		ocv-capacity-celsius = <25>;
+		ocv-capacity-table-0 = <4328000 100>, <4266000 95>, <4208000 90>,
+			<4154000 85>, <4102000 80>, <4062000 75>, <3992000 70>,
+			<3960000 65>, <3914000 60>, <3870000 55>, <3840000 50>,
+			<3818000 45>, <3800000 40>, <3784000 35>, <3770000 30>,
+			<3756000 25>, <3736000 20>, <3714000 16>, <3696000 13>,
+			<3690000 11>, <3689000 10>, <3688000 9>, <3686000 8>,
+			<3682000 7>, <3670000 6>, <3639000 5>, <3592000 4>,
+			<3530000 3>, <3448000 2>, <3320000 1>, <3000000 0>;
+	};
+
 	gpio-keys {
 		compatible = "gpio-keys";
 
@@ -126,6 +144,11 @@ &mdss {
 	status = "okay";
 };
 
+&pm8916_bms {
+	status = "okay";
+	monitored-battery = <&battery>;
+};
+
 &pm8916_pwm {
 	status = "okay";
 	pinctrl-names = "default";
-- 
2.34.1

